linux/check_os_fullname/check_os_fullname.yml

```diff
@@ -38,30 +38,21 @@
             - guest_os_ansible_distribution == 'Flatcar'
             - guest_os_ansible_distribution_ver is version('3760.2.0', '>=')
 
-        - name: "Get hardware version for checking guest info on ESXi {{ esxi_version }}"
-          when: esxi_version is version('8.0.3', '!=')
-          block:
-            - name: "Get ESXi hardware versions"
-              include_tasks: ../../common/esxi_get_hardware_versions.yml
-    
-            - name: "Set fact of latest hardware version for checking guest info on ESXi {{ esxi_version }}"
-              ansible.builtin.set_fact:
-                guest_info_latest_hwv:  "{{ esxi_latest_hardware_version }}"
+        - name: "Get ESXi hardware versions"
+          include_tasks: ../../common/esxi_get_hardware_versions.yml
 
-        - name: "Set hardware version for checking guest info on ESXi {{ esxi_version }}"
+        - name: "Set facts of hardware versions for checking VM guest info"
           ansible.builtin.set_fact:
-            guest_info_latest_hwv: 22
-          when: esxi_version is version('8.0.3', '==')
+            vm_base_hardware_version: "{{ vm_hardware_version_num }}"
```

Comment:
I think here can be make it consistent in the names of these variables, 'hwv', 'hardware_version', or 'hw_version', and add comments for the meaning of each one.

Reply:
Updated. Thanks!